apps/todomvc-remix/app/routes/api.auth.callback.google.tsx

```diff
@@ -0,0 +1,61 @@
+import { LoaderFunction, redirect } from "@remix-run/node";
+import { keelAccessCookie, keelRefreshCookie } from "~/cookies.server";
+
+const clientId = process.env.GOOGLE_CLIENT_ID;
+const clientSecret = process.env.GOOGLE_CLIENT_SECRET;
+
+if (!clientId) {
+  throw new Error("Missing GOOGLE_CLIENT_ID");
+}
+
+if (!clientSecret) {
+  throw new Error("Missing GOOGLE_CLIENT_SECRET");
+}
+
+export const loader: LoaderFunction = async ({ request }) => {
+  const { searchParams } = new URL(request.url);
+  const params = Object.fromEntries(searchParams.entries());
+
+  const data = await getIdTokenWithCode(params.code);
```

Comment:
Should you not being doing something here to validate the csrf token?

If out of scope of the example then maybe just a comment here saying it should be handled?

Reply:
Great catch. Sorted!